shell/store/index.js

```diff
@@ -960,6 +960,8 @@ export const actions = {
 
     const filters = getters['prefs/get'](NAMESPACE_FILTERS)?.[id];
 
+    commit('setProduct', VIRTUAL);
```

Comment:
This is similar to one of the issues with the workspace filter (and we've added a similar workaround)

The authentication middleware runs after `store.app.router.beforeEach` `setProduct`, so anything within the middleware that requires the product has to manually set it.

Reply:
Nice suggestion! Now the `setProduct` in authentication middleware is like the workspace filter.